keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -213,12 +211,12 @@ public boolean equals(Object o) {
 
         Vertex<?> vertex = (Vertex<?>) o;
 
-        return uuid == vertex.uuid;
+        return Objects.equals(this.id, vertex.id);
```

Comment:
`Objects.equals` returns true if both ids are null. Is this what we want?

Reply:
Someone would have to explicitly write null to the ID to make that happen - for now - I'll just make the ID final so that this definitely can't happen - ok?